executor/executor.go

```diff
@@ -427,18 +429,43 @@ func (executor *Executor) execute(
 
 				if runningPods == targetPodCount {
 					msg = statusMap[status]
+					result = proto.DecisionExecutionStatusSucceed
 					break
 				}
 			}
 		}
 
-		executor.logger.Infof(ctx, msg, "time: ", time.Now(), " .... ", start)
+		//rollback in case of failed to restart all pods
+		if runningPods < targetPodCount {
+
+			msg = statusMap[kv1.PodFailed]
```

Comment:
are we sure that `statusMap` will always have `kv1.PodFailed` value? this will panic if we tried to access unkown value

Reply:
yes, if you checked the above code you will find it